crates/router/src/services/authentication.rs

```diff
@@ -666,6 +666,15 @@ where
             metrics::PARTIAL_AUTH_FAILURE.add(1, &[]);
         };
 
+        let profile_id =
```

Comment:
use Headermap and this function defined on it `get_id_type_from_header_if_present`